nox/_options.py

```diff
@@ -225,19 +233,43 @@ def _posargs_finalizer(
     return posargs[dash_index + 1 :]
 
 
+def _python_completer(
+    prefix: str, parsed_args: argparse.Namespace, **kwargs: Any
+) -> Iterable[str]:
+    global_config = parsed_args
+    module = load_nox_module(global_config)
+    manifest = discover_manifest(module, global_config)
+    return (
+        session.func.python
+        for session, _ in manifest.list_all_sessions()
+        if session.func.python
+    )
+
+
 def _session_completer(
     prefix: str, parsed_args: argparse.Namespace, **kwargs: Any
-) -> list[str]:
+) -> Iterable[str]:
     global_config = parsed_args
     global_config.list_sessions = True
     module = load_nox_module(global_config)
     manifest = discover_manifest(module, global_config)
     filtered_manifest = filter_manifest(manifest, global_config)
     if isinstance(filtered_manifest, int):
         return []
-    return [
+    return (
         session.friendly_name for session, _ in filtered_manifest.list_all_sessions()
-    ]
+    )
+
+
+def _tag_completer(
+    prefix: str, parsed_args: argparse.Namespace, **kwargs: Any
+) -> Iterable[str]:
+    global_config = parsed_args
+    module = load_nox_module(global_config)
+    manifest = discover_manifest(module, global_config)
+    return itertools.chain.from_iterable(
```

Comment:
Like before

Maybe it's better change to
```python
return itertools.chain.from_iterable(
        filter(None, (session.tags for session, _ in manifest.list_all_sessions()))
    )
 ```

Reply:
49634fcb169165f1a822a97272ee6628d258e553